House-of-glass/src/Contexts/Cart/CartContext.jsx

```diff
@@ -0,0 +1,174 @@
+import { createContext, useContext, useReducer, useEffect } from "react";
+import axios from "axios";
+import { cartReducer } from "./CartReducer";
+import Toast from "components/Toast/Toast";
+
+const CartContext = createContext();
+
+const useCart = () => useContext(CartContext);
+
+const CartProvider = ({ children }) => {
+  const initialCartState = {
+    cart: [],
+    cartCount: 0,
+    cartSum: 0,
+    cartDiscount: 0,
+  };
+  const [cartState, cartDispatch] = useReducer(cartReducer, initialCartState);
+
+  const token = localStorage.getItem("Token");
+  const header = {
+    headers: {
+      authorization: token,
+    },
+  };
+
+  const getCart = async () => {
+    try {
+      const response = await axios.get("/api/user/cart", header);
+      if (response.status === 200 || response.status === 201) {
+        cartDispatch({ type: "GET_CART", payload: response?.data?.cart });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  useEffect(() => {
+    getCart();
+  }, []);
+
+  const addToCart = async (product) => {
+    if (cartState.cart.some((item) => item._id === product._id)) {
+      incrementCart(product);
+    } else {
+      try {
+        const response = await axios.post(
+          "/api/user/cart",
+          { product },
+          header
+        );
+
+        if (response.status === 201) {
+          cartDispatch({ type: "ADD_TO_CART", payload: response?.data?.cart });
+        }
+        Toast({
+          type: "success",
+          message: `${product.name} will be seeing you soon!`,
+        });
+      } catch (error) {
+        Toast({
+          type: "error",
+          message: "Something broke. Where's my glasses.",
+        });
+        console.log(error);
+      }
+    }
+  };
+
+  const deleteFromCart = async (product) => {
```

Comment:
All cart item functions can be extracted in a separate util file

Reply:
Yes, will do that in an upcoming branch. Thank You.